webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/SystemVariableService.scala

```diff
@@ -676,16 +690,49 @@ object ComputeSchedule {
         val actualStartMinute = startMinute % minutesFreq
         val mins              = Range(actualStartMinute, 60, minutesFreq) // range doesn't return the end range
         // val mins = for ( min <- 0 to 59; if ((min%minutesFreq) == actualStartMinute) ) yield { min }
-        Right(mins.map("\"Min" + "%02d".format(_) + "\"").mkString(", "))
+        Right((LocalTime.of(0, actualStartMinute), mins.map("\"Min" + "%02d".format(_) + "\"").mkString(", ")))
 
       case _ =>
         // hour is not 0, then we don't have minutes
         val actualStartHour = startHour % hoursFreq
         val hours           = Range(actualStartHour, 24, hoursFreq)
         val minutesFormat   = "Min" + "%02d".format(startMinute)
-        Right(hours.map("\"Hr" + "%02d".format(_) + "." + minutesFormat + "\"").mkString(", "))
+        Right(
+          (
+            LocalTime.of(actualStartHour, startMinute),
+            hours.map("\"Hr" + "%02d".format(_) + "." + minutesFormat + "\"").mkString(", ")
+          )
+        )
+    }
+
+  }
+
+  /*
+   * An algorithm to get a splaytime in seconds from a node ID and a execution interval.
+   *
+   * There is no guarantee about the distribution of splay, and you must use random input
+   * to have a change to get something barely random (uuids are ok for that, and it's the
+   * use case).
+   */
+  def computeSplayTime(nodeId: String, interval: Duration): Duration = {
+    if (nodeId.isEmpty) Duration.ofSeconds(0)
+    else {
+      val bigInt       = BigInt.apply(nodeId.getBytes(StandardCharsets.UTF_8))
+      val secondPerDay = 86400
+      Duration.ofSeconds(bigInt.mod(interval.getSeconds).mod(secondPerDay).toLong)
     }
+  }
+
+  /*
+   * Convert an agent schedule into the starttime string which include the splaytime
+   */
+  def getSplayedStartTime(nodeId: String, startTime: LocalTime, interval: Duration): LocalTime = {
+    startTime.plus(computeSplayTime(nodeId, interval))
+  }
 
+  def formatStartTime(startTime: LocalTime): String = {
+    // the withNano(0) is to only print "hh:mm:ss", not "hh:mm:ss.sss"
+    startTime.withNano(0).format(DateTimeFormatter.ISO_TIME)
```

Comment:
Since we use a `LocalTime` we will have no offset, but `DateTimeFormatter.ISO_TIME` could theoretically add one... `DateTimeFormatter.ISO_LOCAL_TIME` would have been more suited but it's also fine as is !